src/constants/mod.rs

```diff
@@ -32,4 +32,14 @@ pub fn account_size_bits<Fp:PrimeFieldParams>() -> usize {
 
 
 //fist 12 bytes from keccak256("ZeroPool")
-pub const ENCRYPTION_NONCE: [u8;12] = [0x5b, 0xbd, 0xff, 0xc6, 0xfe, 0x73, 0xc4, 0x60, 0xf1, 0xb2, 0xb8, 0x5d];
\ No newline at end of file
+pub const ENCRYPTION_NONCE: [u8;12] = [0x5b, 0xbd, 0xff, 0xc6, 0xfe, 0x73, 0xc4, 0x60, 0xf1, 0xb2, 0xb8, 0x5d];
+
+/// Size of prealloced buffer for shared secrets decryption.
+/// It's enough for shared secrets with 10 or less keys.
+pub const SHARED_SECRETS_HEAPLESS_SIZE: usize = 32 * 10 + 16;
```

Comment:
Is it possible to increase this value to maximum amount of keys (128)? In this way every decryption will be in place. Or it will inefficient to allocate too much memory for this, considering mobile devices?

Reply:
Yes, I think it is not worth it. The case when it would be helpful is quite rare but it requires allocating ~4kb on stack constantly.